lib/builtins/deploy-delegates/lambda-deployer/helper.js

```diff
@@ -239,36 +239,40 @@ function _updateLambdaFunction(reporter, lambdaClient, options, callback) {
     const zipFile = fs.readFileSync(zipFilePath);
     const functionName = deployState.lambda.arn;
     let { revisionId } = deployState.lambda;
-    lambdaClient.updateFunctionCode(zipFile, functionName, revisionId, (codeErr, codeData) => {
+    lambdaClient.updateFunctionCode(zipFile, functionName, revisionId, function updateConfiguration(codeErr, codeData) {
         if (codeErr) {
             return callback(codeErr);
         }
         reporter.updateStatus(`Update a lambda function (${functionName}) in progress...`);
-        const { runtime } = userConfig;
-        const { handler } = userConfig;
-        revisionId = codeData.RevisionId;
-        lambdaClient.updateFunctionConfiguration(functionName, runtime, handler, revisionId, (configErr, configData) => {
-            if (configErr) {
-                return callback(null, {
-                    isAllStepSuccess: false,
+        if (codeData.State === 'Pending') {
+            setTimeout(() => lambdaClient.getFunction(functionName, updateConfiguration), 1000);
```

Comment:
The data structure returned by `getFunction` and `updateFunctionCode` is different. The `State` property doesn't exist in the former.

Reply:
Good call out. It's Configuration.State for getFunction. Typescript would be nice! lol

Reply:
I am actually almost done building a PR for this change. I won't be super pretty but it will fix this issue.

Reply:
Awesome, thanks a bunch. Is it a lot different than what I have here?

Reply:
Quite a bit 😄 The create function workflow also need to wait for the proper status and both `State` and `LastUpdateStatus` properties need to be checked.

Anyway, the code is pretty much done. Ideally it should have been refactored to ES6 but I tried to stick to the current format. Testing is showing good results so far. I just need to update the testing suite.